### PR TITLE
Make metadata and name mandatory

### LINKED ISSUE #1036: Make metadata.name mandatory
The metadata.name field is optional in the current entity schema. That has also led to the metadata blob being non-mandatory too. This is an unnecessary complication.

Make the metadata blob mandatory, and make the name mandatory. Adjust both the TS types and the validation accordingly.

### PR DESCRIPTION
**Pull request by freben**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/catalog-model/src/entity/policies/SchemaValidEntityPolicy.test.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ describe('SchemaValidEntityPolicy', () => {
   // metadata
   //
 
-  it('accepts missing metadata', async () => {
+  it('rejects missing metadata', async () => {
     delete data.medatata;
     await expect(policy.apply(data)).resolves.toBe(data);
```
Comment (nikek) [inline packages/catalog-model/src/entity/policies/SchemaValidEntityPolicy.test.ts:85]: shouldn't this also throw?

Comment (nikek) [reply packages/catalog-model/src/entity/policies/SchemaValidEntityPolicy.test.ts:85]: also typo in `medatata`

Comment (freben) [reply packages/catalog-model/src/entity/policies/SchemaValidEntityPolicy.test.ts:85]: Yep the typo made it not throw, haha :$

Comment (freben) [reply packages/catalog-model/src/entity/policies/SchemaValidEntityPolicy.test.ts]: fixed!

**Review by Rugvip — APPROVED** [review]
Nice to get rid of metadata!

**Pull request merged**